GitHub Action fails when repo has no package-lock.json

## Problem

The xfg GitHub Action fails with the error:

```
##[error]Some specified paths were not resolved, unable to cache dependencies.
```

This happens because the action has npm caching hardcoded in `action.yml`:

```yaml
- name: Setup Node.js
  uses: actions/setup-node@v6
  with:
    node-version: "20"
    cache: "npm"
    cache-dependency-path: "**/package-lock.json"
```

When the consuming repository doesn't have a `package-lock.json` file, the glob pattern fails to match and the action errors out.

## Suggested Fix

Either:
1. Remove npm caching entirely (xfg is installed globally via npx anyway)
2. Make caching conditional - only enable if lock files exist
3. Use a try/catch or `continue-on-error` for the setup step

## Reproduction

Use the action in any repo without a `package-lock.json`:

```yaml
- uses: anthony-spruyt/xfg@v1.10.1
  with:
    config: ./config.yaml
    github-token: ${{ secrets.TOKEN }}
```

--- a/.github/workflows/trivy-scan.yaml
+++ b/.github/workflows/trivy-scan.yaml
@@ -23,7 +23,7 @@ jobs:
       - name: Setup Node.js
         uses: actions/setup-node@v6
         with:
-          node-version: "20"
+          node-version: "24"
           cache: "npm"
 
       - name: Install dependencies

--- a/action.yml
+++ b/action.yml
@@ -52,9 +52,7 @@ runs:
     - name: Setup Node.js
       uses: actions/setup-node@v6
       with:
-        node-version: "20"
-        cache: "npm"
-        cache-dependency-path: "**/package-lock.json"
+        node-version: "24"
 
     - name: Install GitLab CLI
       if: ${{ inputs.gitlab-token != '' }}

--- a/docs/ci-cd/github-actions.md
+++ b/docs/ci-cd/github-actions.md
@@ -107,7 +107,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: "20"
+          node-version: "24"
       - run: npm install -g @aspruyt/xfg
       - run: xfg --config ./config.yaml
         env:
@@ -149,7 +149,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: "20"
+          node-version: "24"
       - run: npm install -g @aspruyt/xfg
       - run: xfg --config ./configs/${{ matrix.config }}
         env:
@@ -173,7 +173,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: "20"
+          node-version: "24"
       - run: npm install -g @aspruyt/xfg
       - run: xfg --config ./config.yaml --dry-run
         env:

--- a/docs/use-cases.md
+++ b/docs/use-cases.md
@@ -58,7 +58,7 @@ files:
             - uses: actions/checkout@v4
             - uses: actions/setup-node@v4
               with:
-                node-version: "20"
+                node-version: "24"
             - run: npm ci
             - run: npm test
 
@@ -215,7 +215,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: "20"
+          node-version: "24"
       - run: npm install -g @aspruyt/xfg
       - run: xfg --config ./config.yaml
         env:

